src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,898 @@
+                 TestStep("6d", "TH reads from the DUT the OperationalState attribute"),
+                 TestStep("6e", "Manually put the device in the Paused(0x02) operational state"),
+                 TestStep("6f", "TH reads from the DUT the OperationalState attribute"),
+                 TestStep("6g", "Manually put the device in the Error(0x03) operational state"),
+                 TestStep("6h", "TH reads from the DUT the OperationalState attribute"),
+                 TestStep(7, "TH reads from the DUT the OperationalError attribute"),
+                 TestStep("7a", "Manually put the device in the NoError(0x00) error state"),
+                 TestStep("7b", "TH reads from the DUT the OperationalError attribute"),
+                 TestStep("7c", "Manually put the device in the UnableToStartOrResume(0x01) error state"),
+                 TestStep("7d", "TH reads from the DUT the OperationalError attribute"),
+                 TestStep("7e", "Manually put the device in the UnableToCompleteOperation(0x02) error state"),
+                 TestStep("7f", "TH reads from the DUT the OperationalError attribute"),
+                 TestStep("7g", "Manually put the device in the CommandInvalidInState(0x03) error state"),
+                 TestStep("7h", "TH reads from the DUT the OperationalError attribute")
+                 ]
+        return steps
+
+    async def TEST_TC_OPSTATE_BASE_2_1(self, endpoint=1):
+        cluster = self.test_info.cluster
+        attributes = cluster.Attributes
+
+        self.init_test()
+
+        # STEP 1: Commission DUT to TH (can be skipped if done in a preceding test)
+        self.step(1)
+
+        # STEP 2: TH reads from the DUT the PhaseList attribute
+        self.step(2)
+        if self.pics_guard(self.check_pics(("%s.S.A0000" % self.test_info.pics_code))):
+            phase_list = await self.read_expect_success(endpoint=endpoint,
+                                                        attribute=attributes.PhaseList)
+            if phase_list is not NullValue:
+                phase_list_len = len(phase_list)
+                asserts.assert_less_equal(phase_list_len, 32,
+                                          "PhaseList length(%d) must be less than 32!" % phase_list_len)
+
+        # STEP 3: TH reads from the DUT the CurrentPhase attribute
+        self.step(3)
+        if self.pics_guard(self.check_pics(("%s.S.A0001" % self.test_info.pics_code))):
+            current_phase = await self.read_expect_success(endpoint=endpoint,
+                                                           attribute=attributes.CurrentPhase)
+            if (phase_list == NullValue) or (not phase_list):
+                asserts.assert_true(current_phase == NullValue, "CurrentPhase(%s) should be null" % current_phase)
+            else:
+                asserts.assert_true(0 <= current_phase and current_phase < phase_list_len,
```

Comment:
In the future consider using the assert_less_equal and assert_less tests instead.

Reply:
fixed